### DISCUSSION TITLE
VisiData 4Ever

### DISCUSSION THREAD

**Discussion by ajkerrigan** [saulpw/visidata#1533] [❤️]
Following some discussion with @saulpw and @anjakefala - this is a follow-up to #595 since that one has gotten quite long. _(That's a good thing!)_

**Comment by saulpw** [comment]
There were many responses in favor of VisiData from [this Twitter thread](https://twitter.com/simonw/status/1572285367382061057), in which @simonw asks:

> If someone gives you a CSV file with 100,000 rows in it, what tools do you use to start exploring and understanding that data?

which is exactly what VisiData was built for :)

**Reply by saulpw** [reply]
> Up to 1M rows, visidata is my go to tool. Quick custom filtering and histograms give a good sense of the data within a minute or so.
- [Yasharth](https://twitter.com/ychdry/status/1572398052815802368)

**Comment by frosencrantz** [comment]
I like it when I see these comments in which people reference VisiData on Hacker News and it is not the main point of their comment.   It is just a part of how they performed the task, like you might mention nailing piece of wood together as part of assembly.  This is a tool in their tool belt, that is a high compliment.

[pasabagi](https://news.ycombinator.com/user?id=pasabagi) writes on [Bank of England expects UK to fall into longest ever recession](https://news.ycombinator.com/item?id=33455646)
> Are we looking at the same graph? Aside from 2012-14, it consistently shows the UK trailing the pack in terms of growth, often by multiple percentage points. Bear in mind, even a single percentage point is quite a big deal: the chinese economy was 'roaring' at an average of 4% growth per year.
I loaded up the csv in **visidata**, and the graph you sent me shows the UK trails the EU growth average by half a percentage point since 2008. Against a relatively well-performing economy like germany, it trails by 0.94% over the same period.

OR 

[farresito](https://news.ycombinator.com/user?id=farresito) writes  on: [Integrating variants in 42k autism cases identifies mutations in new genes](https://news.ycombinator.com/item?id=32520180)
> Getting your whole genome sequenced is around 200-300$. Nebula or Dante labs offer that. Once you receive the data, which is going to be a VCF file, you have to annotate it with something like Ensembl VEP, which adds useful information to analyze it. After that, you can download a list of genes associated with autism from SFARI and extract the variants that you have in those genes. You remove the common variants (say those that have more than 5% frequency) and focus on the coding variants (the ones that affect the exome). From there on, the interesting part begins. Depending on the kind of variant, you can get certain conclusions or certain others. There are tools that predict the pathogenicity of certain variants; other variants you know almost certainly that they render the resulting protein non-functional, so you have to take a closer look at the gene and see if having a single copy of the gene is enough. And so on.
It's not complicated at all and a lot of it can be done with basic Linux tools, like grep, and visualized with CSV viewers like **Visidata**. That's just a basic explanation. If you end up getting your genome sequenced and want to analyze it, let me know in a comment.

**Reply by saulpw** [reply]
Thanks, @frosencrantz !  How do you find these?  I have a watch on 'visidata' and I don't think these turned up.

**Reply by frosencrantz** [reply]
This is what I check every so often: 
https://hn.algolia.com/?dateRange=all&page=0&prefix=false&query=visidata&sort=byDate&type=all

**Reply by aborruso** [reply]
> Thanks, @frosencrantz ! How do you find these? I have a watch on 'visidata' and I don't think these turned up.

You have also RSS feed:

- https://hnrss.org/newest?q=visidata
- https://hnrss.org/newcomments?q=visidata

**Reply by frosencrantz** [reply]
Now I want RSS loading support in VisiData beyond what the XML loader does.

**Comment by saulpw** [comment]
David Thiel wrote up [a Mastodon thread](https://mastodon.lol/@det/109643412313479438) about their [ucluster VisiData plugin](https://github.com/stanfordio/ucluster) for grouping similar text to detect spam:

> The plugin implementation is due to the fact that while our “prod” is generally BigQuery with various frontends, I do much of my initial analysis (and sometimes all of it) using API-emitted JSONL in VisiData (because it is awesome), running it through a local enrichment pipeline. Part of this pipeline can be ucluster — here, I’ve run the fuzzy-cluster command on the “text” column, which returns the numeric cluster to which the tweet text belongs, or -1 if it couldn’t decide.

**Comment by saulpw** [comment]
From https://twitter.com/LovelyDataCZ/status/1600599361620217856:

> [translation] What do you think, is there a tool that is equal to Excel or even better? A tool that can spreadsheet, is easy to use, is fast and does not cost a penny? [https://www.lovelydata.cz/blog/visidata-lepsi-nez-excel/](https://www-lovelydata-cz.translate.goog/blog/visidata-lepsi-nez-excel/?_x_tr_sl=tr&_x_tr_tl=ja&_x_tr_hl=en&_x_tr_pto=wapp)

**Comment by frosencrantz** [comment]
From https://chaos.social/@scy/110396676007788276

> How could I miss out on [#VisiData](https://fosstodon.org/tags/VisiData) for so long? This might become my new favorite [#CLI](https://fosstodon.org/tags/CLI) tool.
>
> If you do _anything_ with data and enjoy working in the terminal, check it out. It can
>
> • provide a [#TUI](https://fosstodon.org/tags/TUI) for viewing and editing data in [#CSV](https://fosstodon.org/tags/CSV), [#Excel](https://fosstodon.org/tags/Excel), [#SQLite](https://fosstodon.org/tags/SQLite), [#JSON](https://fosstodon.org/tags/JSON), [#YAML](https://fosstodon.org/tags/YAML) & [#XML](https://fosstodon.org/tags/XML) files and quite a few more
> • sort, filter, join and edit that data, across files and across formats
> • convert between the formats (interactively or not)
> • record & play macros
> • be scripted in [#Python](https://fosstodon.org/tags/Python)

**Comment by anjakefala** [comment]
![Screenshot from 2023-06-19 22-46-40](https://github.com/saulpw/visidata/assets/7489659/e95f2ec0-698b-47b6-a6c5-b95b299dbb7a)

https://discord.com/channels/880915750007750737/880915984481939466/1120227410588225548

> just discovered that vd can draw shapefiles and i cannot stop smiling. omg thank you
> very, very slightly life-changing lol

> I'm doing some work with TIGER data, nothing super exciting but being able to look at stuff in vd to spot-check what I am doing is very useful (usually I'd load it into qgis and that takes a while)

**Comment by anjakefala** [comment]
> Can confirm, I consider visidata to be the gold standard for data exploration.

> I have largely given up on spreadsheets, replacing them with relational databases. And on top of being a world class tool for understanding the data for import, visidata makes for a pretty great query pager. 

- somat https://news.ycombinator.com/item?id=38894276

**Comment by larstiq** [comment]
Originally posted a slightly different version on the `#visidata` irc channel but decided to go for a wider audience on Mastodon (https://mathstodon.xyz/@larstiq/112284974755111300)

> One little [#visidata](https://mathstodon.xyz/tags/visidata) trick that brought me joy yesterday: sort a dataset by one column (e.g. modified date), select a range of interesting rows, sort by another column (e.g name or description) and then use `{`/`}` to jump to the previous/next selected rows and inspect them in the context of their neighboring rows.  One of the most delightful pieces of software I use on a regular basis.

**Comment by pratyushmittal** [comment]
I love VisiData. It inspires me to do things with love and thought.

I use VisiData to browse through any dump of data. Sometimes logs on a remote server. Sometimes "my activity" dump on Spotify or Facebook. Things I wouldn't have attempted before.

Most useful thing about VisiData is it can digest *anything*: xmls, jsons, CSVs...

Yesterday, I asked my accountant to send me a dump of our company accounts. Our accounting software works only on Windows. I needed something to view that data on Mac. I asked him to export data in csv.

VisiData did magic. I was able to search through entries, "filter them", "create new sheets", "do joins", even create running totals (window feature) – all in few keystrokes.

Macros have been super useful. Being able to run them as commands (press space) is unimaginable.

Thanks a lot for creating VisiData. I wish more people knew it and used it.

VisiData is 100x duck db, pandas, excel - put together.

**Comment by frosencrantz** [comment]
VisiData is at 8,037 stars on GitHub!   Congratulation!  It passed 8K sometime over the last month.

**Comment by saulpw** [comment]
From @pequiste:

> I demonstrated the power of vd today at work. I needed to filter a list for a referendum and we have thousands couples sharing an email account. So dupes are okay as long as they aren't the same person. And sometimes they are because there is a typo in the name and they got added twice.
>
> So freq on the emails to find the dupes, full join with the list against itself, compute the levenshtein distance, sort in ascending order.
>
> About a dozen cases to fix, that was very easy.
>
> The others wanted to do it with excel and were stumped.

**Comment by libbysparx** [comment]
I only learned about VisiData yesterday and so far I've basically just used it for having my Breath of the Wild spreadsheets and checklists handy on my latest run, and I am loving it to bits ^_^  I'm kind of a CLI/TUI slut and usually my screen is dominated by a cluster of ~~haphazard~~ *very assiduously organized* terminator windows so this is my absolute jam of the moment.

Haven't bothered to make a custom theme for it yet because the ascii8 theme is pretty much perfect and could make the driest inventory slog pleasant and satisfying to me <3

**Reply by anjakefala** [reply]
We are Zelda fans here, and are delighted to be part of your gaming experience!

**Comment by jjmca** [comment]
I use Visidata at work every single day. Fantastic tool that just works. Thank you so much.

**Comment by aszenz** [comment]
Learning visidata now, after using sql and malloy for data transformations, it's really refreshing to find something that does so much with so little effort. 

I think adoption could increase if it's not marketed as a terminal spreadsheet since it's so much more.

**Reply by midichef** [reply]
I'd be happy to hear some of your favorite use cases, or favorite commands to use. I am considering writing up a set of short demo scripts showing the power of visidata. If you can explain what visidata makes easy, that is difficult with SQL and malloy, that would help me think about what to put in a demo.

**Reply by aszenz** [reply]
Visidata makes it easier to breakdown complex data transformations into steps since you see results immediately and pipelining is just a sheet away unlike sql which requires complex ctes / subqueries
Also unlike sql I don't have to visualize the result set completely upfront, I can be more iterative.
Visidata is better for spotting quirks in data since you see every intermediate steps output.

**Comment by frosencrantz** [comment]
There is some high praise for VisiData in a simple Hacker News comment from [Simon Willison](https://github.com/simonw).  High praise in what is said, and who is saying it.  Simon is the co-creator of Django, creator of Datasette, and a Python  Software Foundation board member. He blogs about data journalism and LLMs.

The actual comment: `(saulpw is the author of VisiData, and it's a marvelous piece of software.)`
https://news.ycombinator.com/item?id=46267464  part of a thread started by Saul: http://news.ycombinator.com/item?id=46267365

**Reply by midichef** [reply]
That is indeed meaningful praise, coming from Simon Willison, whose work is outstanding. Anyone who uses Visidata should also check out [Datasette](https://datasette.io). It's fantastic for exploring data in the browser. From the Datasette website:

>Datasette is a tool for exploring and publishing data. It helps people take data of any shape or size and publish that as an interactive, explorable website and accompanying API.
>Datasette is aimed at data journalists, museum curators, archivists, local governments, scientists, researchers and anyone else who has data that they wish to share with the world.

**Comment by austrowj** [comment]
Visidata is the GOAT of data exploration for me. Absolutely incredible design. No other software consistently makes me break into a huge smile the way vd does.

I read someone say that doing pretty much anything data is so much faster using vd, _even if_ you have to look up every single command you use. That is 100% accurate.

@saulpw Thank you so much for creating and maintaining vd. That is all.
